http4k-serverless-lambda/src/main/kotlin/org/http4k/serverless/lambda/LambdaFunction.kt

```diff
@@ -1,31 +1,66 @@
 package org.http4k.serverless.lambda
 
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent
 import org.http4k.core.Body
+import org.http4k.core.Filter
 import org.http4k.core.MemoryBody
 import org.http4k.core.Method
 import org.http4k.core.Request
+import org.http4k.core.RequestContexts
 import org.http4k.core.Response
 import org.http4k.core.Uri
+import org.http4k.core.then
 import org.http4k.core.toUrlFormEncoded
+import org.http4k.filter.ServerFilters
 import org.http4k.serverless.BootstrapAppLoader
 
+const val LAMBDA_CONTEXT_KEY = "HTTP4K_LAMBDA_CONTEXT"
+const val LAMBDA_REQUEST_KEY = "HTTP4K_LAMBDA_REQUEST"
+
 /**
  * This is the main entry point for the lambda. It uses the local environment
  * to instantiate the Http4k handler which can be used for further invocations.
  */
 class LambdaFunction(env: Map<String, String> = System.getenv()) {
-    private val app = BootstrapAppLoader(env)
+    private val contexts = RequestContexts()
+    private val app = BootstrapAppLoader(env, contexts)
+    private val initializeRequestContext = ServerFilters.InitialiseRequestContext(contexts)
 
-    fun handle(request: ApiGatewayProxyRequest) = app(request.asHttp4k()).asApiGateway()
+    fun handle(request: APIGatewayProxyRequestEvent, lambdaContext: Context? = null) =
```

Comment:
Is the `Context? = null` here just for the tests? AWS Lambda will always provide this when the function has the second argument.

Reply:
Yes. Some of the tests directly use invoke this without setting up the contexts, but on lambda this will indeed always be there.

Reply:
I've merged it but a couple of questions/changes:

1. Instead of putting the entire `APIGatewayProxyRequestEvent` I'm going to put the use the `ProxyRequestContext` to avoid duplication.